### PR TITLE
tests/network: drop remaining NewRandomVMIWithEphemeralDiskAndUserdata

### PR DESCRIPTION
**Pull request by dankenigsberg**
`tests.NewRandomVMIWithEphemeralDiskAndUserdata` is long and rigid relative to the alternative provided by `libvmi.WithCloudInitNoCloudUserData`. Make the required changes to stop using it in tests/network (mostly reading the return
value of tests.Wait* commands to collect a vmi object with its namespace set)

```release-note
NONE
```

### CODE REVIEW

File: tests/network/vmi_multus.go [hunk 1]
```diff
@@ -672,7 +671,7 @@ var _ = SIGDescribe("[Serial]Multus", func() {
                     ip addr add %s dev ep1
                     ip addr add %s dev ep2
                 `, ep1Cidr, ep2Cidr, ep1CidrV6, ep2CidrV6)
-				agentVMI := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling), userdata)
+				agentVMI := libvmi.NewFedora(libvmi.WithCloudInitNoCloudUserData(userdata, false))
```
Comment (maiqueb) [inline tests/network/vmi_multus.go:674]: IIUC, `agentVMI` was a `NewTestToolingFedora` but would now be a `NewFedora` . Is this change intentional ? If so, would you elaborate on its motivation ?

File: tests/libvmi/cloudinit.go [hunk 2]
```diff
@@ -34,9 +34,11 @@ func WithCloudInitNoCloudUserData(data string, b64Encoding bool) Option {
 		volume := getVolume(vmi, diskName)
 		if b64Encoding {
 			encodedData := base64.StdEncoding.EncodeToString([]byte(data))
+			volume.CloudInitNoCloud.UserData = ""
```
Comment (maiqueb) [inline tests/libvmi/cloudinit.go:37]: I do not understand the use case of running `WithCloudInitNoCloudUserData` twice on the same VMI. 

Could you explain it ?

**Review by maiqueb — COMMENTED** [review]
I Have a couple of questions, and I suspect one of the introduced changes was not intentional.

File: tests/network/vmi_multus.go (hunk 1 above)
Comment (dankenigsberg) [reply tests/network/vmi_multus.go:674]: No, it was not intentional. I'll fix it. However please note that for several months now, the two entities are [one and the same](https://github.com/kubevirt/kubevirt/blob/main/tests/libvmi/factory.go#L37).

File: tests/libvmi/cloudinit.go (hunk 2 above)
Comment (dankenigsberg) [reply tests/libvmi/cloudinit.go:37]: I attempted to explain this in the [commit message](https://github.com/kubevirt/kubevirt/pull/7342/commits/a7f4db59e1d35dc919b3a28a96ebf138a12b276c). 

NewCirros has a default non-empty cloud-init value. To allow the user
override it with a specific user data such as in `libvmi.NewCirros(libvmi.WithCloudInitNoCloudUserData(userData, false))`, we need to ensure the default one is not left behind.

Comment (maiqueb) [reply tests/libvmi/cloudinit.go:37]: Maybe I've expressed myself poorly: I'm interesting in understanding the reason why would someone want to call `WithCloudInitNoCloudUserData`  twice on the same VMI. 

I understand the fix. I just don't imagine myself ever wanting to do what the fix enables.

File: tests/network/vmi_multus.go (hunk 1 above)
Comment (maiqueb) [reply tests/network/vmi_multus.go]: wow. OK. Isn't that something we should somehow fix ? I mean... it's misleading at best. 

Let's say you have functions `NewSalad` and `NewSaladWithLotsOfNuts` . These - for whatever reason - are the same.
If I were allergic to nuts, I would happily consume the `NewSalad`, thinking myself very safe, and then go and have an anaphylaxis attack.

Comment (dankenigsberg) [reply tests/network/vmi_multus.go]: Yes, I found it quite surprising myself. I don't see if this question was raised when https://github.com/kubevirt/kubevirt/commit/36aba3ec0d7feecc379bd157ca3e424c14bf2e48#diff-6744b3e54011de85527b1a5b5deb98fbc05a60aa7576c644315d5dc25ebcb74bL40 was discussed. It is also not clear to me what is the right solution: is it to re-split between vanilla Fedora and Fedora-with-nuts? Or drop all traces of `NewTestToolingFedora` from the code?

I tend for the lazier second option. Anyway, would you not agree that it is unrelated to this tiny PR?

File: tests/libvmi/cloudinit.go (hunk 2 above)
Comment (dankenigsberg) [reply tests/libvmi/cloudinit.go:37]: I don't think that I understand your point, then.

No test should ever call `libvmi.New(WithCloudInitNoCloudUserData(1), WithCloudInitNoCloudUserData(2), ...)`; but my suggested code does this implicitly in `libvmi.NewCirros(libvmi.WithCloudInitNoCloudUserData(userData, false))`

Do you find this solution bad/confusing? Do you have a better idea to let `libvmi.NewCirros()` users modify its default user data?

File: tests/network/vmi_multus.go (hunk 1 above)
Comment (maiqueb) [reply tests/network/vmi_multus.go]: 100 % not related :) 

I actually think the *right* thing (take it with a pinch of salt ...) is to have the 2 of them. But it falls to the user (person writing the tests) to know which to use. I don't think it's too hard to decide: "need tools?" => use the test tooling one. 

But hey, let's discuss this somewhere down the line on a PR dedicated to this.

File: tests/libvmi/cloudinit.go (hunk 2 above)
Comment (maiqueb) [reply tests/libvmi/cloudinit.go:37]: Ah. I now see what you mean. It's been too long since I used any of this; I had to look at the code to realize what "NewCirros has a default non-empty cloud-init value" meant. It's [exactly what it says](https://github.com/kubevirt/kubevirt/blob/4c72454beeb171076f73170cf661f00277d15ae2/tests/libvmi/factory.go#L84). My bad.

I actually think the correct thing to do is again to have 2 types of cirros VMIs - one with, one without user data. `NewCirros` , and `NewCirrosWithCloudInit` . I find that more explicit. 

I'm OK with letting this in and eventually discuss if we want to take that route for all VM OSs, since we're discussing the exact same thing - but for fedora - on https://github.com/kubevirt/kubevirt/pull/7342/#discussion_r824689921.

**Comment by dankenigsberg** [comment]
/test pull-kubevirt-e2e-k8s-1.22-ipv6-sig-network 

No idea why this job is stuck in "Expected — Waiting for status to be reported" for about 24h.

**Comment by dankenigsberg** [comment]
And now it [fails](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7342/pull-kubevirt-e2e-k8s-1.22-ipv6-sig-network/1504015434934915072#1:build-log.txt%3A199) on
```
08:44:27: gazelle: finding module path for import golang.org/x/sys/unix: golang.org/x/sys@v0.0.0-20220315194320-039c03cc5b86: verifying module: golang.org/x/sys@v0.0.0-20220315194320-039c03cc5b86: reading https://goproxy.io/sumdb/sum.golang.org/lookup/golang.org/x/sys@v0.0.0-20220315194320-039c03cc5b86: stream error: stream ID 3; INTERNAL_ERROR; received from peer
```
